pkg/service/connections.go

```
@@ -125,9 +125,30 @@ func (s *Service) TerminateUDPConnectionsToBackend(l3n4Addr *lb.L3n4Addr) error
 
 // backendConnectionHandler is added for dependency injection in tests.
 type backendConnectionHandler struct {
-	logger *slog.Logger
+	sockets.SocketDestroyer
+	logger     *slog.Logger
+	useNetlink bool
+}
+
+func newBackendConnectionHandler(l *slog.Logger, sockTermFilter *filter.SockTermFilterMap) *backendConnectionHandler {
+	return &backendConnectionHandler{
+		SocketDestroyer: &sockets.BPFSocketDestroyer{
+			SockTermFilter: sockTermFilter,
+			Logger:         l,
+		},
+		logger: l,
+	}
 }
 
 func (h backendConnectionHandler) Destroy(filter sockets.SocketFilter) error {
-	return sockets.Destroy(logging.DefaultSlogLogger, filter)
+	err := h.SocketDestroyer.Destroy(filter)
+	if err != nil && !h.useNetlink {
```

This would end up logging the error on kernels < 6.5 that don't have the kfunc - https://github.com/cilium/cilium/blob/main/pkg/service/connections.go#L73.

---

You mean wouldn't end up logging? Yeah, I guess we could add a log message here to better inform the user if the BPF strategy doesn't work instead of just hiding the error.

---

Yup, I had a typo. Error reporting looks better.